app/code/Magento/Customer/Model/Address.php

```diff
@@ -325,10 +325,7 @@ public function getEntityTypeId()
      */
     public function afterSave()
     {
-        $indexer = $this->indexerRegistry->get(Customer::CUSTOMER_GRID_INDEXER_ID);
-        if ($indexer->getState()->getStatus() == StateInterface::STATUS_VALID) {
-            $this->_getResource()->addCommitCallback([$this, 'reindex']);
-        }
+        $this->_getResource()->addCommitCallback([$this, 'reindex']);
```

Comment:
So, we need such change in to places just to get rid of `RecurringData`? Does not sound too good.

Reply:
Triggering a reindex when the index is valid sounds good to you?

Reply:
I have the feeling this PR now starts to fix two things which are not really related. Could it be that the issue @IvanPletnyov is bringing up also happens without this PR?

Reply:
@slackerzz it will be reindexed naturally if the index is not valid I believe.

Reply:
Probably. but then https://travis-ci.org/magento/magento2/jobs/529661515 will fail again i think